tests/ui/rfcs/rfc-2008-non-exhaustive/improper_ctypes/auxiliary/types.rs

```diff
@@ -27,3 +27,14 @@ pub enum NonExhaustiveVariants {
     #[non_exhaustive] Tuple(u32),
     #[non_exhaustive] Struct { field: u32 }
 }
+
+// Note the absence of repr(C): it's not necessary, and recent C code can now use repr hints too.
+#[repr(u32)]
+#[non_exhaustive]
```

Comment:
Wait, does `#[repr(u32)] #[non_exhaustive]` (as opposed to `repr(C)`) also warn without this PR?
In this case I don't see the motivation, new potentially added variants cannot change the ABI.

Reply:
Hm? The "potentially-added variants that might change the ABI" are ones that would change it to a `really_tagged_enum`.

Reply:
Ah, I see, the new variants can add data.
(FWIW, the "really tagged" terminology is not used anywhere in the compiler so it wasn't obvious what it means.)

Reply:
fair enough, I didn't really know what else to call it except by the RFC title. ^^